pkg/client/client.go

```diff
@@ -18,6 +18,8 @@ type Client interface {
 	Create(ctx context.Context, obj runtime.Object) error
 	Get(ctx context.Context, key ObjectKey, objExisting runtime.Object) error
 	Update(ctx context.Context, obj runtime.Object) error
+	Delete(ctx context.Context, obj runtime.Object, opts ...client.DeleteOptionFunc) error
+	List(ctx context.Context, opts *client.ListOptions, list runtime.Object) error
```

Comment:
This is a side-effect of needing to use the client interface in both sets of reconcilers. The catalogsourceconfig's delete reconciler needs to list and delete resources. Would it help for me to mention that I updated the client interface in the 1st commit message?

Reply:
Then shouldn't you be introducing this change in the commit where you start using the two new additions?

Reply:
These are used in this commit. Because we pass in this client interface in each of these methods these new Delete and LIst methods will be used [here](https://github.com/operator-framework/operator-marketplace/pull/198/commits/d59caf31fca43b6ee33649a8da90d55e69a4f615#diff-b0cc6bb0c0e2413e3d6903a35fca00f4R116) and [here](https://github.com/operator-framework/operator-marketplace/pull/198/commits/d59caf31fca43b6ee33649a8da90d55e69a4f615#diff-b0cc6bb0c0e2413e3d6903a35fca00f4R109) for example. The code that uses them didn't need to be updated

Reply:
Ah I see. Please mention the reason for doing so in the commit message.